src/OpenTelemetry.ResourceDetectors.Gcp/GcpResourceDetector.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using Google.Api.Gax;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.ResourceDetectors.Gcp;
```

Comment:
Would it be better if we start this one with the proposal from https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1610?